crates/uv/src/commands/project/lock.rs

```diff
@@ -645,6 +645,7 @@ async fn do_lock(
                 None,
                 resolver_env,
                 python_requirement,
+                workspace.conflicting_groups(),
```

Comment:
To clarify: does this mean conflicting groups must be defined in the workspace root?

Could we instead have each project define its own conflicts, and then remove `package =` from the schema? I don't think it's necessary to allow arbitrary packages to be included in these groups, since we only allow _enabling_ extras for members in the workspace.

Reply:
I think this makes sense to me. From the resolver's perspective, it definitely needs the package name, but from the user's perspective, I can see how it makes sense to keep the conflicting declaration local to where the corresponding extras/groups are themselves defined. And in that context, the package name is implicit.

I do wonder if there exists a conflict that does require the user to specify the package name, but I can't quite think of how that might manifest.

I'll plan to remove `package` from the schema.

Reply:
(But note that we still need `package` in the lock file.)